### DISCUSSION TITLE
Failed to build the project with CMake in Windows: warning MSB8003 and fatal error C1083

### DISCUSSION THREAD

**Discussion by ccartermices** [NVlabs/instant-ngp#984] [General]
Hey! I use CMake to build the project (on Windows, used in developer command prompt), with the command:
instant-ngp$ cmake . -B build
However, it failed:
Here are some of the log:

`-- Selecting Windows SDK version  to target Windows 10.0.19044.
CMake Error at D:/cmake/share/cmake-3.24/Modules/CMakeDetermineCompilerId.cmake:739 (message):
  Compiling the CUDA compiler identification source file
  "CMakeCUDACompilerId.cu" failed.

  Compiler:

  Build flags:

  Id flags: --keep;--keep-dir;tmp -v

  The output was:

  1

 .NET Framework  Microsoft (R)  16.9.0+57a23d249

  C:\Program Files (x86)\Microsoft Visual
  Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(501,5):
**MSBuild warning MSB8003: The property-name property is not defined. Some build tools may not be found.** 
  [E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj]

  InitializeBuildStatus:

“Debug\CompilerIdCUDA.tlog\unsuccessfulbuild”，“AlwaysCreate”。

  AddCudaCompileDeps:

    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\bin\HostX64\x64\cl.exe /E /nologo /showIncludes /TP /D__CUDACC__ /D__CUDACC_VER_MAJOR__=11 /D__CUDACC_VER_MINOR__=7 /D_MBCS /I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" /I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin" /I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" /I. /FIcuda_runtime.h /c E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CMakeCUDACompilerId.cu

  CudaBuildCore:

    Compiling CUDA source file CMakeCUDACompilerId.cu...
    “E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\x64\Debug”。
    cmd.exe /C "C:\Users\cc\AppData\Local\Temp\tmpbc646da89acf46f3b8d263484ef1ea68.cmd"
    "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin\nvcc.exe" -gencode=arch=compute_52,code=\"sm_52,compute_52\" --use-local-env -ccbin "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\bin\HostX64\x64" -x cu   -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include"  -G   --keep-dir x64\Debug  -maxrregcount=0  --machine 64 --compile -cudart static -v -g  -D_MBCS -Xcompiler "/EHsc /W0 /nologo /Od /FdDebug\vc142.pdb /FS /Zi /RTC1 /MDd " -o E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\x64\Debug\CMakeCUDACompilerId.cu.obj "E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CMakeCUDACompilerId.cu"

    E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA>"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin\nvcc.exe" -gencode=arch=compute_52,code=\"sm_52,compute_52\" --use-local-env -ccbin "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\bin\HostX64\x64" -x cu   -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include"  -G   --keep-dir x64\Debug  -maxrregcount=0  --machine 64 --compile -cudart static -v -g  -D_MBCS -Xcompiler "/EHsc /W0 /nologo /Od /FdDebug\vc142.pdb /FS /Zi /RTC1 /MDd " -o E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\x64\Debug\CMakeCUDACompilerId.cu.obj "E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CMakeCUDACompilerId.cu"
    #$ _NVVM_BRANCH_=nvvm
    #$ _SPACE_=
    #$ _CUDART_=cudart
    #$ _HERE_=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin
    #$ _THERE_=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin
    #$ _TARGET_SIZE_=
    #$ _TARGET_DIR_=
    #$ _TARGET_SIZE_=64
    #$ _WIN_PLATFORM_=x64
    #$ TOP=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin/..
    #$ NVVMIR_LIBRARY_DIR=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin/../nvvm/libdevice
    #$ PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin/../nvvm/bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin/../lib;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\bin\HostX64\x64;;;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\tools;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\ide;C:\Program Files (x86)\HTML Help Workshop;;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin;C:\WINDOWS\Microsoft.NET\Framework64\v4.0.30319\;C:\WINDOWS\Microsoft.NET\Framework\v4.0.30319\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin;C:\WINDOWS\Microsoft.NET\Framework\v4.0.30319\;;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\libnvvp;C:\Program Files\PlasticSCM5\server;C:\Program Files\PlasticSCM5\client;C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\intel64\compiler;D:\python3.10.5\Scripts\;D:\python3.10.5\;D:\xshell\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Common Files\Autodesk Shared\;D:\matlab2020b\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;D:\quicktime\QTSystem\;D:\cmake\bin;C:\Program Files\NVIDIA Corporation\Nsight Compute 2022.2.1\;C:\Users\cc\AppData\Local\Microsoft\WindowsApps;D:\ae2015\goodbyegreenscreen\1.0;C:\Program Files\GoodbyeGreenscreen;D:\ae2020\goodbyegreen\depend;%GUN_HOME%\bin;
    #$ INCLUDES="-IC:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin/../include"
    #$ LIBRARIES=  "/LIBPATH:C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin/../lib/x64"
    #$ CUDAFE_FLAGS=--sdk_dir ""
    #$ PTXAS_FLAGS=
    CMakeCUDACompilerId.cu
    #$ cl.exe @"C:\Users\cc\AppData\Local\Temp/tmpxft_00004cac_00000000-9.res" > "C:/Users/cc/AppData/Local/Temp/tmpxft_00004cac_00000000-8_CMakeCUDACompilerId.cpp1.ii"
    CMakeCUDACompilerId.cu

  C:\Program Files (x86)\Microsoft Visual
  Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\crtdefs.h(10):
**fatal error C1083: Cannot open compiler intermediate file: “corecrt.h”: No such file**

  [E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj]

    # --error 0x2 --

  C:\Program Files (x86)\Microsoft Visual
  Studio\2019\Community\MSBuild\Microsoft\VC\v160\BuildCustomizations\CUDA
  11.7.targets(790,9): error MSB3721: command“"C:\Program Files\NVIDIA GPU
  Computing Toolkit\CUDA\v11.7\bin\nvcc.exe"
  -gencode=arch=compute_52,code=\"sm_52,compute_52\" --use-local-env -ccbin
  "C:\Program Files (x86)\Microsoft Visual
  Studio\2019\Community\VC\Tools\MSVC\14.28.29910\bin\HostX64\x64" -x cu
  -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include"
  -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -G
  --keep-dir x64\Debug -maxrregcount=0 --machine 64 --compile -cudart static
  -v -g -D_MBCS -Xcompiler "/EHsc /W0 /nologo /Od /FdDebug\vc142.pdb /FS /Zi
  /RTC1 /MDd " -o
  E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\x64\Debug\CMakeCUDACompilerId.cu.obj
  "E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CMakeCUDACompilerId.cu"”，return code is 
  2。
  [E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj]

“E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj”(CudaBuildCore
  ) - failed。

“E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj”
  - failed。

  failed。

  “E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj”(defalut target)
  (1) ->

  (PrepareForBuild target) ->

    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(501,5): MSBuild warning MSB8003 : WindowsSDKDir is not defined. Some build tools may not be found. 。 [E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj]

  “E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj”(defalut target)
  (1) ->

  “E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj”(CudaBuildCore
  target) (1:2) ->

  (CudaBuildCore target) ->

    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\crtdefs.h(10): fatal error C1083: Cannot open compiler intermediate file: “corecrt.h”: No such file or directory [E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj]
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\BuildCustomizations\CUDA 11.7.targets(790,9): error MSB3721: 命令“"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin\nvcc.exe" -gencode=arch=compute_52,code=\"sm_52,compute_52\" --use-local-env -ccbin "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\bin\HostX64\x64" -x cu   -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include"  -G   --keep-dir x64\Debug  -maxrregcount=0  --machine 64 --compile -cudart static -v -g  -D_MBCS -Xcompiler "/EHsc /W0 /nologo /Od /FdDebug\vc142.pdb /FS /Zi /RTC1 /MDd " -o E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\x64\Debug\CMakeCUDACompilerId.cu.obj "E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CMakeCUDACompilerId.cu"”已退出，返回代码为 2。 [E:\instant-ngp-master\build\CMakeFiles\3.24.1\CompilerIdCUDA\CompilerIdCUDA.vcxproj]

      1 warnning
      2 errors

used time 00:00:02.00

Call Stack (most recent call first):
  D:/cmake/share/cmake-3.24/Modules/CMakeDetermineCompilerId.cmake:6 (CMAKE_DETERMINE_COMPILER_ID_BUILD)
  D:/cmake/share/cmake-3.24/Modules/CMakeDetermineCompilerId.cmake:48 (__determine_compiler_id_test)
  D:/cmake/share/cmake-3.24/Modules/CMakeDetermineCUDACompiler.cmake:307 (CMAKE_DETERMINE_COMPILER_ID)
  CMakeLists.txt:11 (project)

-- Configuring incomplete, errors occurred!
See also "E:/instant-ngp-master/build/CMakeFiles/CMakeOutput.log".
See also "E:/instant-ngp-master/build/CMakeFiles/CMakeError.log".`

**Comment by roman-kudinov** [comment]
I have the same error, Visual Studio 2022, CUDA 11.8, Windows 10

As a workaround I've updated 
D:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\BuildCustomizations\CUDA 11.8.props

and added references to Universal CRT headers and libs to CudaToolkitIncludeDir and CudaToolkitcLibDir:

        `<CudaToolkitIncludeDir Condition="'$(CudaToolkitIncludeDir)' == ''">$(CudaToolkitDir)include;C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt</CudaToolkitIncludeDir>`

and 

        `<CudaToolkitLibDir Condition="'$(CudaToolkitLibDir)' == '' AND Exists('$(CudaToolkitDir)lib\x64')">$(CudaToolkitDir)lib\x64;C:\Program Files (x86)\Windows Kits\10\Lib\10.0.19041.0\ucrt\x64</CudaToolkitLibDir>`
